src/draw_target.rs

```diff
@@ -398,8 +398,13 @@ impl std::ops::DerefMut for DrawStateWrapper<'_> {
 impl Drop for DrawStateWrapper<'_> {
     fn drop(&mut self) {
         if let Some(orphaned) = &mut self.orphan_lines {
-            orphaned.extend(self.state.lines.drain(..self.state.orphan_lines_count));
-            self.state.orphan_lines_count = 0;
+            orphaned.extend(
```

Comment:
This commit has a lot going on. Would you mind trying to split it into smaller commits that make a more minimal single logical change?